File: LeetCode/Binary Search/practice.py
def search(nums, target):
    left , right = 0, len(nums)-1

    while left <= right:
        mid = (left + right) // 2

        if target > nums[mid]:
            left = mid + 1
        elif target < nums[mid]:
            right = mid - 1
        else:
            return mid
    
    return -1

print(search([-1,0,3,5,9,12], 9))

File: LeetCode/Binary Search/practice.js
function search(nums, target){
    let left = 0
    let right = nums.length - 1

    while (left <= right){
        mid = Math.trunc((left + right) / 2)

        if(target > nums[mid]){
            left = mid + 1
        } else if (target < nums[mid]){
            right = mid - 1
        } else {
            return mid
        }
    }
    return "Number doesn't exist"
}

console.log(search([-1,0,3,5,9,12], 2))
